dangerzone/updater/releases.py

```diff
@@ -0,0 +1,227 @@
+import json
+import platform
+import sys
+import time
+from abc import ABC
+from dataclasses import dataclass
+
+# The "|" syntax for type unions was introduced with Python 3.10
+# So we use Union instead as we still require Python 3.9
+from typing import Optional, Tuple, Union
+
+import markdown
+import requests
+from packaging import version
+
+from .. import container_utils, util
+from ..settings import Settings
+from . import errors, log
+from .signatures import (
+    DEFAULT_PUBKEY_LOCATION,
+    get_remote_digest_and_logindex,
+)
+
+# Check for updates at most every 12 hours.
+UPDATE_CHECK_COOLDOWN_SECS = 60 * 60 * 12
+
+GH_RELEASE_URL = (
+    "https://api.github.com/repos/freedomofpress/dangerzone/releases/latest"
+)
+REQ_TIMEOUT = 15
+
+
+@dataclass
+class ReleaseReport:
+    """
+    A new Github Release, a new sandbox image (or both) have been detected
+    """
+
+    version: Optional[str] = None
+    changelog: Optional[str] = None
+    container_image_bump: bool = False
+
+    @property
+    def new_github_release(self) -> bool:
+        return self.version is not None
+
+    @property
+    def is_empty(self) -> bool:
+        return (
+            self.version is None
+            and self.changelog is None
+            and not self.container_image_bump
+        )
```

Comment:
It's kinda weird to have an `is_empty()` method here, only to use it later on in order to switch an instance of this class with an `EmptyReport`. I guess my question is, do we really need two separate classes for these report types?

Reply:
It's a matter of style I guess. The previous version where we've had everything in one class felt harder to read and to reason about. I prefer to have an error type separate, and then for the empty report I guess it's like easier to use: you check for the type and you know what to do.

I understand why it might feel strange to trade one for the other in the implementation, but at the same time I like the API it exposes.

Another way would be to have intermediate variables and then decide what to do with it, but I dislike it even more I think, hence why I went with this.

Let me know if it's a blocker for you, and I can try to change it to something else.